core/src/main/java/nl/nn/adapterframework/senders/XsltSender.java

```diff
@@ -172,7 +195,21 @@ public String sendMessage(String correlationID, String message, ParameterResolut
 //				log.debug(getLogPrefix()+" prc.inputsource ["+prc.getInputSource()+"]");
 //			}
 
-			stringResult = transformerPool.transform(inputMsg, parametervalues);
+			TransformerPool poolToUse = transformerPool;
+			
+			
+			if(!StringUtils.isEmpty(styleSheetNameSessionKey) && prc.getSession().get(styleSheetNameSessionKey) != null) {
+				String styleSheetNameToUse = prc.getSession().get(styleSheetNameSessionKey).toString();
+
+				if(!dynamicTransformerPoolMap.containsKey(styleSheetNameToUse)) {
+					dynamicTransformerPoolMap.put(styleSheetNameToUse, poolToUse = TransformerPool.configureTransformer(getLogPrefix(), getClassLoader(), getNamespaceDefs(), getXpathExpression(), styleSheetNameToUse, getOutputType(), !isOmitXmlDeclaration(), getParameterList()));
```

Comment:
Hier moet je in plaats van getNamespaceDefs(), getXpathExpression() en getOutputType() steeds null invullen, omdat het anders gaat conflicteren.
Het lijkt me goed om voor die situatie ook een test toe te voegen, dus: styleSheetNameSessionKey i.c.m. xpathexpression